spring-integration-core/src/main/java/org/springframework/integration/channel/NullChannel.java

```diff
@@ -163,6 +165,7 @@ public void onComplete() {
 
 	private TimerFacade sendTimer() {
 		if (this.successTimer == null) {
+			Assert.state(this.metricsCaptor != null, "metricsCaptor not must not be null");
```

Comment:
Same here with a `@SuppressWarnings("NullAway") // Dataflow analysis limitation`.
Same reason as in the `AbstractMessageChannel`.